nilearn/glm/second_level/second_level.py

```diff
@@ -274,9 +274,13 @@ def _get_con_val(second_level_contrast, design_matrix):
         else:
             raise ValueError("No second-level contrast is specified.")
     if not isinstance(second_level_contrast, str):
-        con_val = second_level_contrast
-        if np.all(con_val == 0):
-            raise ValueError("Contrast is null")
+        con_val = np.array(second_level_contrast)
+        if np.all(con_val == 0) or len(con_val) == 0:
+            raise ValueError(
+                "Contrast is null. Second_level_contrast must be a valid "
+                "contrast vector, a list/array of 0s and 1s, a string, or a "
+                "string expression."
+            )
```

Comment:
@Remi-Gau one thing I'm unsure about is this part. I think it should rather be:

```python
        if np.all(con_val == 0) or len(con_val) == 0:
            raise ValueError(
                "Contrast is null."
            )
        elif <some other check>:
            raise ValueError(
                "Second_level_contrast must be a valid "
                "contrast vector, a list/array of 0s and 1s, a string, or a "
                "string expression."
```

Then I'm wondering how to check for valid contrast. Should we allow both a contrast vector summing to 0 or an array with one nonzero number like in the check `(np.nonzero(second_level_contrast)[0]).size != 1` in the removed function `_get_contrast`?

Reply:
Not sure how much we want to check things here, that would be a case where I would let the users see if the contrast vector "makes sense".

Reply:
There is no reason to restrict contrasts. Any non-null contral is valid.